https://bugs.gentoo.org/318287: Fix buffer bound checks causing an
abort when -f commandline parameter is passed.

--- a/src/cpufreqd.h
+++ b/src/cpufreqd.h
@@ -54,6 +54,12 @@
 #define DEFAULT_VERBOSITY	3
 
 #define MAX_STRING_LEN		255
-#define MAX_PATH_LEN		512
+
+#ifdef HAVE_LIMITS_H
+#include <limits.h>
+#define MAX_PATH_LEN PATH_MAX
+#else
+#define MAX_PATH_LEN 512
+#endif
 
 #endif /* __CPUFREQD_H__ */
